compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -34,4 +40,110 @@ luci::CircleNode *CloneNodeLet<CN::OPQR>::visit(const luci::CircleReshape *node)
   return cloned;
 }
 
+namespace sinf
+{
+
+/**
+ * @note  CircleReshape has new shape info in two places: 2nd input and attribute.
+ *        This shape inference uses shape from input 'shape' node when it's constant.
+ *        If not, shape will be from node itself. shape from attribute is not used.
+ *
+ * TODO Change this policy when not appropriate
+ */
+loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
+{
+  LOGGER(l);
+
+  const loco::DataType S32 = loco::DataType::S32;
+
+  bool is_input_known = true;
+  auto input_tensor = dynamic_cast<luci::CircleNode *>(node->tensor());
+  for (uint32_t axis = 0; axis < input_tensor->rank(); ++axis)
+  {
+    if (!input_tensor->dim(axis).known())
+    {
+      is_input_known = false;
+      break;
+    }
+  }
+
+  loco::TensorShape shape_by_input;
+  {
+    LUCI_ASSERT(node->shape(), "2nd input shape() should not be nullptr");
+
+    // Only support node's shape() is CircleConst with S32
+    // TODO support other node with other types
+    auto const_shape_node = dynamic_cast<luci::CircleConst *>(node->shape());
+    if (const_shape_node != nullptr)
+    {
+      LUCI_ASSERT(const_shape_node->dtype() == S32, "Only support int32 CircleConst");
+
+      shape_by_input.rank(const_shape_node->size<S32>());
+
+      for (uint32_t axis = 0; axis < shape_by_input.rank(); ++axis)
+      {
+        shape_by_input.dim(axis) = const_shape_node->at<S32>(axis);
+        if (const_shape_node->at<S32>(axis) < 0)
+          shape_by_input.dim(axis).unset();
+      }
+    }
+    else
+    {
+      // We use shape from the node itself
+      shape_by_input = own_shape(node);
+    }
+  }
+
+  loco::TensorShape shape_by_attr;
+  {
+    shape_by_attr.rank(node->newShape()->rank());
+
+    for (uint32_t axis = 0; axis < shape_by_attr.rank(); ++axis)
+    {
+      shape_by_attr.dim(axis) = node->newShape()->dim(axis);
+    }
+  }
+
+  if (!(shape_by_input == shape_by_attr))
+  {
+    INFO(l) << "CircleReshape: Two new shape information mismatched : " << std::endl;
+    INFO(l) << "   shape_by_input : " << shape_by_input << std::endl;
+    INFO(l) << "   shape_by_attr : " << shape_by_attr << std::endl;
+  }
+
+  loco::TensorShape output_shape = shape_by_input;
+
+  // One of the dimensions can have special value -1, meaning its actual value should be inferred.
```

Comment:
have you changed any logic here?
existing code logic is based with infer with known shape and do infer as much as possible with unknown shapes.

title if this PR is to do shape infer with dynamic shape.
I think this logic should be changed.
please discuss with your SAFFY mantor about how to.